### PR TITLE
Clean up base lists

### LINKED ISSUE #392: Audit base lists
## Context

Check that base lists are not too opinionated.

- @yaili to confirm base `<dl>` style
- Remove list-inline and list -patterns

## Checklist

- [ ] Browser testing
- [ ] Accessibility testing
- [ ] Sass linting
- [ ] Update documentation
- [ ] Convert sizing units to rems
- [ ]  Ensure only elements are styled directly

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Simplified base list styles
- Updated docs to include ordered, unordered and nested lists

## QA

- Sanity check code
- Sync and test with vf.io

## Details

Fixes #392

### CODE REVIEW

File: scss/_base_lists.scss [hunk 1]
```diff
@@ -1,27 +1,24 @@
 @mixin vf-lists {
+  
   ol,
   ul {
-    margin-left: 20px;
-    margin-bottom: 20px;
-  }
+    margin-left: 1rem;
+    margin-bottom: 1rem;
+    padding-left: 1rem;
```
Comment (anthonydillon) [inline scss/_base_lists.scss]: Are we adding left padding to lists now?

File: scss/_base_lists.scss [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
 
-  ol ol,
-  ul ul,
-  ol ul,
-  ul ol {
-    margin-bottom: 0;
-  }
+    ul,
+    ol, {
+      margin-bottom: 0;
```
Comment (anthonydillon) [inline scss/_base_lists.scss]: I think we want lists to have the same bottom margin as p tags. Otherwise they will look bad in text.

**Comment by anthonydillon** [comment]
Looks good with a few comments in code :+1:

File: scss/_base_lists.scss (hunk 1 above)
Comment (barrymcgee) [inline scss/_base_lists.scss]: For base lists - yes.

File: scss/_base_lists.scss (hunk 2 above)
Comment (barrymcgee) [inline scss/_base_lists.scss]: This rule applies only to nested lists.

**Pull request merged**